### DISCUSSION TITLE
Internal Linking Solution

### DISCUSSION THREAD

**Discussion by cbratschi** [directus/directus#18537] [Feature Requests]
### Summary

Looking for a solution to add internal links to other or the same collection in Directus. Currently only plain URLs are supported by the TinyMCE dialog.

### Basic Example

Directus shall support internal stateless linking in WYSIWYG fields by introducing the following URL schema: directus://<collection name>/<item id>. An dialog shall be added which allows the selection of the collection and its items. It is up to the user to parse and convert such URIs.

### Motivation

Internal linking is a fundamental feature. Hard coding URLs breaks the statelessness of Directus: URLs only work on the target frontend system they were written for e.g. staging URLs are different than production URLs. This approach always easy migration of a staging system to production.

### Detailed Design

URI scheme:

directus://<collection name>/<item id>

Where both the collection name and the item ID are URL encoded.

### Must Have:
- Add TinyMCE link dialog which allows to use internal linking.
- Generate directus:// URIs.

### Should Have:
- Easy way to navigate to the referred item.

### Could Have:
- Link validation.

### Drawbacks

Users have to convert URIs using regex to target URLs manually.

### Alternatives

None.

### Adoption Strategy

This is an incremental feature. There might be an opt-in or opt-out to keep the current handling.

### Unresolved Questions

Do extensions already support this?

**Comment by AxeemHaider** [comment]
I find out the solution, actually it's already there not documented but if you write url which starts from `/` it consider as internal url.

**Comment by cbratschi** [comment]
@AxeemHaider you mean using relative or absolute paths? What I am asking for is on a different level: it should be possible to directly link to any item of a collection without knowing the path in advance.

**Comment by nickvenne** [comment]
Agreed, It would be a cool feature if we could in the wysiwyg editor link to a item in some collection x to some field y. So if you ever change that y field. all wysiwyg editor links automatically update. Currently a huge burden for blogs posts etc..

**Comment by rijkvanzanten** [comment]
Heya! Thanks for opening this feature request!

This feature request has received over **15 votes** from the community. This means we'll move this feature request to the Under Review state! 

The Core team will schedule a meeting to review this request as soon as possible. The discussion will then be approved or denied. You may or may not be invited to join this meeting with the core team.

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Reply by greensomes** [reply]
Has this request been approved? Is there any movement on implementing this feature?

**Comment by lotarbo** [comment]
This feature is available in other cms
I hope to see something like this here soon, I'm looking forward to it.
https://payloadcms.com/docs/rich-text/slate#relationship-element
https://www.sanity.io/guides/portable-text-internal-and-external-links

**Comment by formfcw** [comment]
Small note: With the release of `v1.8.0` the extension [Flexible Editor](https://github.com/formfcw/directus-extension-flexible-editor) provides a solution for internal linking :rocket:


https://github.com/user-attachments/assets/fa518ab3-6630-47cf-885f-b171dc1efa71

**Reply by MentalGear** [reply]
This is great - thank you!

**Comment by MentalGear** [comment]
It's essential for CMS editors to have the capability to link to other internal pages easily. The absence of this standard feature in Directus CMS significantly hampers its usability and potential for widespread adoption.
